compiler/parser.go

```diff
@@ -589,10 +589,47 @@ func (l *treeNodeListener) EnterInnerTxnAssign(ctx *gen.InnerTxnAssignContext) {
 	l.node = node
 }
 
+func (l *treeNodeListener) EnterInnerTxnArrayAssign(ctx *gen.InnerTxnArrayAssignContext) {
+	field := ctx.TXNARRAYFIELD().GetText()
+	node := newArrayAssignInnerTxnNode(l.ctx, l.parent, field)
+	listener := newExprListener(l.ctx, node)
+	ctx.Expr().EnterRule(listener)
+	exprToPush := listener.getExpr()
+	node.value = exprToPush
```

Comment:
nit: this value does not get any special treatment so can go into `node.children`. But there are bunch of similar cases that need to be addressed so nit.